sfepy/terms/terms_multilinear.py

```diff
@@ -746,8 +740,8 @@ def make_function(self, texpr, *args, diff_var=None):
 
         if einfo.eargs is None:
             einfo.eargs = [
-                ExpressionArg.from_term_arg(arg, self, self.expr_cache)
-                for arg in args
+                ExpressionArg.from_term_arg(arg, arg_name, self, self.expr_cache)
+                for arg, arg_name in zip(args, self.arg_types)
```

Comment:
This does not work, because the `make_function()` arguments can be (and are) different from the term arguments.

Reply:
Yes, you are right. My intention is to avoid the second arguments check in `make_function()`. The new version should work. The tests pass.

Now, the names of ndarray `args` can be specified as:  

```
self.make_function('ij,i,j', (mat1, 'material_1'), (mat2, 'material_2'), ...)
```

Reply:
`arg_pos` argument removed. All "non-standard" arguments will be treated with tuples `(arg, arg_name)`.